### ISSUE TITLE
issues with parallel in confint?

### ISSUE THREAD

**Issue by sacool** [glmmTMB/glmmTMB#356]
First of all, my apologies if this is either a trivial issue with my code, or the wrong place to bring this up, but I _think_ what I'm trying to do should show different behaviour than what I'm observing.

I've fit a model with `glmmTMB` on my macbook, and in trying to calculate likelihood profile confidence intervals, I wanted to use something like:

`confint(final_mod, parm=c(2:4), method='profile', parallel='multicore', ncpus=2)`

The confidence intervals are calculated, but **I'm suspicious that they were not calculated in parallel**; using `top -o cpu` in the command line to see what processes are using significant cpu power, I could only see one `R` process running with ~100 %CPU

On the other hand, if I do:

`profile(final_mod, parm=c(3,4), parallel='multicore', ncpus=2)`

I get something similar to what I expect, _i.e._ two `R` processes running at full steam.

I'm using `R version 3.5.0`, `glmmTMB_0.2.1.0` on macOS 10.13.4, on a machine with a 2.6 GHz Intel Core i5 processor (4 logical cores and 2 physical cores).

If there's something obvious that I'm missing, my apologies and go ahead and close this, but I thought I would bring it up in case there is something wrong here.

**Comment by koenvanbenthem** [comment]
I observed the same behavior. To me it seems that the ncpus argument is not passed to profile.

https://github.com/glmmTMB/glmmTMB/blob/04cd11a88b74fbb38fbed59e9c99271145fce8d2/glmmTMB/R/methods.R#L753-L755

(it seems that it is no longer part of the ..., since it is a predefined argument of confint too)

Hence, I suppose this should be changed to:
```
pp <- profile(object, parm = parm, level_max = level, 
                  parallel = parallel, ncpus=ncpus,
                  ...)
```
When I implement this, top seem to imply an actual parallel calculation. I didn't dare changing the code myself, because I'm somewhat worried to whether it might break other cases (e.g. when not wanting to do the calculation in parallel). I hope that identifying the line at least fastens solving this issue.

**Comment by mmaechler** [comment]
Adding the  `ncpus = ncpus`  is definitely what should be done there.
I think the "obvious" logic must be to always pass `ncpus` when you pass or use `parallel` in that parts of the code.

**Comment by koenvanbenthem** [comment]
I have now made this change on a separate branch:
[confint-parallel-fix](/glmmTMB/glmmTMB/tree/confint-parallel-fix)

I did some quick checks and it seems that it is now running in parallel. If there are no objections, this can now be merged into the master branch. (but additional testing is always welcome of course).

**Comment by mebrooks** [comment]
Fixed by #361
